Filter summary window results

cage2 opened the pull request:
Hi!

A friend of mine gave me that idea to filter the entries shown in the summary window, i could not resist! ;-)

I quickly tested this feature and (also given it is simple), in my opinion should work. At this time the parenthesis can not be used as search text but i think this could be fixed using escape and i hope to fix that in the next days.

Essentially before a summary window is shown the user is asked for a query and the database of annotations is filtered against this query; the results ,if any, are shown in the window.

The syntax for this simple query language can be found in the docstring of ` annotate-summary-filter-db`.
The user can disable the query mechanism using customizable variable `annotate-summary-ask-query`.


Bye!
C.

Commit:
- used a better regex when checking if a file is in info format.

Commit:
- enabled filtering of annotation database when a summary window is shown.

bastibe commented on annotate.el:
```diff
@@ -1243,7 +1255,8 @@ sophisticated way than plain text"
                                (let* ((file-contents     (file-contents))
                                       (has-info-p        (string-match "info" filename))
-                                      (has-separator-p   (string-match "" file-contents))
+                                      (has-separator-p   (string-match "\\(\\)?\\(\\)?$"
```
This belongs to the other pull request, right?

bastibe commented:
To be honest, this is somewhat beyond my capacities to review. But I'll take it on faith, no problem.

Could you include a few code comments in the longer functions to make them easier to read? I'd love to understand what you are doing, this is cool stuff!

cage2 commented:
On Mon, Nov 25, 2019 at 12:00:42AM -0800, Bastian Bechtold wrote:
> bastibe commented on this pull request.
>
>
>
> > @@ -1243,7 +1255,8 @@ sophisticated way than plain text"
>                                (t
>                                 (let* ((file-contents     (file-contents))
>                                        (has-info-p        (string-match "info" filename))
> -                                      (has-separator-p   (string-match "" file-contents))
> +                                      (has-separator-p   (string-match "\\(\\)?\\(\\)?$"
>
> This belongs to the other pull request, right?

True, i messed with the branches, sorry! :(

Bye!
C.

Commit:
- added 'cl-' prefix to 'remove-if';
- parenthesis can be escaped when filtering notes;
- fixed error message;
- added more comments, especially for the filtering annotation related
  functions.

cage2 commented:
On Mon, Nov 25, 2019 at 12:03:45AM -0800, Bastian Bechtold wrote:

Hi Bastian!

>
> To be honest, this is somewhat beyond my capacities to review. But I'll take it on faith, no problem.

Thank you!

> Could you  include a few  code comments  in the longer  functions to
> make them easier to read? I'd love to understand what you are doing,
> this is cool stuff!

To be honest i feel i would be  arrogant if i would try to explain the
CS stuff behind  this code (parsing, context free grammar  etc.), i am
not good  enough. So i  have added comments  to explain just  what the
code does. I tried hard to make the commennts useful as much as i can,
given my bad english too! :-)

Bye!
C.

Commit:
- minor fix in comments.

bastibe commented:
> To be honest i feel i would be arrogant if i would try to explain the CS stuff behind this code (parsing, context free grammar etc.), i am not good enough. So i have added comments to explain just what the code does. I tried hard to make the comments useful as much as i can, given my bad english too! :-) Bye! C.

What you added is absolutely perfect! Thank you very, very much! I am not a very experienced lisp programmer, and your comments help me enormously to make sense of the code. Great work!

cage2 commented:
On Tue, Nov 26, 2019 at 01:14:24AM -0800, Bastian Bechtold wrote:

>  What you added is absolutely perfect!  Thank you very, very much! I
> am not a very experienced lisp programmer, and your comments help me
> enormously to make sense of the code. Great work!

Thank you, i  am always happy to  be useful with my  work!  But please
remember that  i am  learning elisp  from the code  you wrote  so, you
helped me too! :)

Bye!
C.

bastibe commented:
As far as I am concerned, you can merge when you're ready.

Commit:
- changed escaping mechanism;
- changed regex for checking if a file is in info format.

cage2 commented:
On Wed, Nov 27, 2019 at 12:36:28AM -0800, Bastian Bechtold wrote:

Hi Bastian!

> As far as I am concerned, you can merge when you're ready.

I changed the way to do escaping of strings (now the software uses the
usual quotation mark delimited string like  "not".

Also i changed the regular expression for info file checking as in the
other message.

If the latter is ok tomorrow evening i am going to merge everything.

Bye!
C.

bastibe commented:
Cool!

cage2 commented:
On Thu, Nov 28, 2019 at 12:00:51AM -0800, Bastian Bechtold wrote:
> Cool!

Ok! Merging and waiting for bug reports! :D :D :D

Bye!
C.
